### ISSUE TITLE
Request for Apollo Guidance Computer assembly (AGC/AEA) Syntax Highlight

### ISSUE THREAD

**Issue by GoogleCodeExporter** [alqxyz/google-code-prettify#81]
```
For the 40th anniversary of the Apollo moon landing we have transcribed the
actual Command Module code (i.e. Comanche054) and Lunar Module code (i.e.
Luminary099) from scanned images to actual source text again which can be
compiled and run on the yaAGC (see http://virtualagc.googlecode.com) which
is an emulator that can run this original code. The simulation provides
also simulated peripherals to interact with the machine (e.g. DSKY and
Telemetry).

To enable Syntax highlighting for this assembly code I have attached a
Prettify extension that covers the assembly of both machines.

The following are links to the example assembly code:
http://code.google.com/p/virtualagc/source/browse/trunk/Comanche055/CM_BODY_ATTI
TUDE.s

http://code.google.com/p/virtualagc/source/browse/trunk/Luminary099/LUNAR_LANDIN
G_GUIDANCE_EQUATIONS.s

http://code.google.com/p/virtualagc/source/browse/trunk/FP8/FP8.s

```

Original issue reported on code.google.com by `ohommes@gmail.com` on 9 Jun 2009 at 1:18

Attachments:
- [lang-apollo.js](https://storage.googleapis.com/google-code-attachments/google-code-prettify/issue-81/comment-0/lang-apollo.js)

**Comment by GoogleCodeExporter** [comment]
```
Please excuse my ignorance, but when is the deadline on this?  What file 
extension
are you planning on using for the files?  '.s'?

The code.google guys need to do a manual push before any code hosted there will 
work
with changes to the root of this project.
```

Original comment by `mikesamuel@gmail.com` on 12 Jun 2009 at 6:13
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
The deadline I am trying to aim for is to have this enabled before July 20th 
2009. 
Because that is the day of the anniversary of the Apollo 11 moon landing.
Our code project also plans to have an anniversary release in place for the 
emulator 
to run the original code just before or on this date.

We have had so far extensive cooperation from the MIT museum (who provided us 
the 
scanned code from Apollo 11 which we transcribed and verifyed with original 
CSUM's) 

The file extension will be '.s'

I really appreciate your time and support in this matter and I look forward to 
seeing the online syntax high lighting.
```

Original comment by `ohommes@gmail.com` on 13 Jun 2009 at 11:48

**Comment by GoogleCodeExporter** [comment]
```
Hmm.  The '.s' extension is used for a lot of scheme source code files, and so 
if the
lang-lisp handler were loaded at the same time as the apollo one, the second one
loaded would fail with an exception.

I don't know quite how google.code decides which language extensions to load, 
but
I'll ask if they might be able to use one for the apollo code, and the lisp one 
for
other projects.

I noticed that your patch registers two extensions : ['apollo', 's'].  If they 
don't
do that, could you release it using the '.apollo' extension?
```

Original comment by `mikesamuel@gmail.com` on 13 Jun 2009 at 3:11

**Comment by GoogleCodeExporter** [comment]
```
The 'apollo' is not really an extension I put it in because in this file I 
actually 
combine already the op-codes of both the Apollo Guidance Computer aswell as the 
Abort Electronics Assembly computer. To change the extension would require some 
dicussion in the development group since all the tools we have including the 
YUL 
compiler uses this extension at the moment. If the syntax highlighting could be 
customized per development project by picking which languages are used that 
would be 
the best solution. For now I keep my fingers crossed.
```

Original comment by `ohommes@gmail.com` on 13 Jun 2009 at 11:54

**Comment by GoogleCodeExporter** [comment]
```
After some limited discussion we are willing to change the '.s' extension to 
'.agc'
for the AGC assembly and '.aea' for the AEA assembly code. This of course would
require us to change our tools on our side and re-submit two separate Prettify
language extension files.

Another option that we can adopt if possible for googlecode is to add a tag in 
the
first line like: #!yaYUL for the AGC assembly code or #!yaLEMAP for the AEA 
assembly
code. These tags would denote the name of our compiler.

We also welcome other suggestions if you have any additional ones to accomplish 
this.
If we need to change the '.s' extension then we might not have this all 
completed
before the 20th of July but will attempt this if this is the preferred option.

So in summary for the two extension we would submit two language extension 
files:
lang-agc.js and lang-aea.js

Let me know your thoughts.
```

Original comment by `ohommes@gmail.com` on 16 Jun 2009 at 1:44

**Comment by GoogleCodeExporter** [comment]
```
It wouldn't require two language extension files for the prettifier.  The last
argument to PR_registerLangHandler is a list of language extensions, so it 
would just
look like PR_registerLangHandler(..., ['aea', 'agc']);.   See lang-lisp.js for
example which includes a bunch of lisp variants including scheme.

I'll follow up with the google code guys to see how they go about figuring which
language name to use for a file.  Maybe there's some magic SVN attribute that 
affects it.
```

Original comment by `mikesamuel@gmail.com` on 16 Jun 2009 at 1:18

**Comment by GoogleCodeExporter** [comment]
```
Okay I'll update and reattach a new version when you get the final word from the
google code guys. I see you prefer a single language extension file so lets 
keep it
like that. Look forward to hearing the feedback soon so in our project we can 
start
planning and outlining the work for the extension change with regards to our 
compiler
toolset.
```

Original comment by `ohommes@gmail.com` on 16 Jun 2009 at 2:59

**Comment by GoogleCodeExporter** [comment]
```
Any feedback from google-code? Let me know if I need to do anything to expedite
things on my side.
```

Original comment by `ohommes@gmail.com` on 23 Jun 2009 at 3:25

**Comment by GoogleCodeExporter** [comment]
```
They need to make the changes on their end.  I'll bug them about it tomorrow.
```

Original comment by `mikesamuel@gmail.com` on 24 Jun 2009 at 7:04

**Comment by GoogleCodeExporter** [comment]
```
They should have that done sometime this week.  Do you have a final version of
lang-apollo.js that I can check into trunk?
```

Original comment by `mikesamuel@gmail.com` on 30 Jun 2009 at 6:59

**Comment by GoogleCodeExporter** [comment]
```
It will be great to see this work. If they can work with the '.s' extension 
then the
version you have is the latest. I had proved it for several days before 
uploading.
However if you need the .aea and .agc extension then I need to send you a new
version. Will the current '.s' extension work? (this would be the least amount 
of
work for our team but I have discussed this and are prepared to change the 
extension
if required)
```

Original comment by `ohommes@gmail.com` on 30 Jun 2009 at 9:54

**Comment by GoogleCodeExporter** [comment]
```
It looks like the extension checked into http://code.google.com/p/google-code-
prettify/source/browse/trunk/src/lang-apollo.js only registers "apollo" as a 
file extension. Should this be 
updated to include "agc" and "aea" as file extensions as well?
```

Original comment by `nathaniel+legacy@google.com` on 13 Jul 2009 at 10:37

**Comment by GoogleCodeExporter** [comment]
```
Yes the apollo extension was just to see if it worked. This definitely needs to 
be
extended to use the agc and aea. Do you want me to make the update? Or will you 
add
.agc and .aea extension?
```

Original comment by `ohommes@gmail.com` on 13 Jul 2009 at 11:33

**Comment by GoogleCodeExporter** [comment]
```
I'll add agc and aea.
```

Original comment by `mikesamuel@gmail.com` on 14 Jul 2009 at 12:41

**Comment by GoogleCodeExporter** [comment]
```
Added
```

Original comment by `mikesamuel@gmail.com` on 16 Jul 2009 at 2:04

**Comment by GoogleCodeExporter** [comment]
```
When can I create a test in the SVN repository? Should it be active already?
```

Original comment by `ohommes@gmail.com` on 16 Jul 2009 at 2:55

**Comment by GoogleCodeExporter** [comment]
```
The highlighting for .aea and .agc should go live in a few hours. You may see 
incorrect highlighting now; but you 
should see correct highlighting in a few hours.
```

Original comment by `nathaniel+legacy@google.com` on 16 Jul 2009 at 5:37

**Comment by GoogleCodeExporter** [comment]
```
The highlighting is now live. .apollo, .aea, and .agc files should now be 
correctly highlighted.
```

Original comment by `nathaniel+legacy@google.com` on 20 Jul 2009 at 5:54

**Comment by GoogleCodeExporter** [comment]
```
I have gone ahead and made the changes to the names of our AGC and AEA source 
files,
so that they are named *.agc and *.aea instead of *.s.  The highlighting seems 
to
work great on the AGC files, but I don't actually see any highlighting in the 
AEA
files.  (For what it's worth, there are only two of the latter at the moment:
FP6/FP6.aea and FP8/FP8.aea.)  Of course, it may be some confusion on my part.

And of course, thanks!

```

Original comment by `rburkey2...@gmail.com` on 21 Jul 2009 at 6:45

**Comment by GoogleCodeExporter** [comment]
```
Those are comparatively large files (nearly five thousand lines each). I 
suspect you're hitting some sort of 
constraint that disables syntax highlighting for such large files, and I'll 
look into it more deeply in the coming 
days.
```

Original comment by `nathaniel+legacy@google.com` on 22 Jul 2009 at 6:30

**Comment by GoogleCodeExporter** [comment]
```
Indeed, Project Hosting currently has a three thousand line limit on syntax 
highlighting. If you'd like this raised, 
please file an issue at http://code.google.com/p/support/issues/entry.
```

Original comment by `nathaniel+legacy@google.com` on 22 Jul 2009 at 3:37

**Comment by GoogleCodeExporter** [comment]
```
Thanks, I will.

```

Original comment by `rburkey2...@gmail.com` on 22 Jul 2009 at 5:13

**Comment by GoogleCodeExporter** [comment]
Original comment by `mikesamuel@gmail.com` on 14 Aug 2009 at 5:51
- Changed state: **Fixed**
